### PR TITLE
[RFC] fixes for solaris

### PR DESCRIPTION
**Pull request by 4nd3r**
what do you think, is that crazy enough?

### CODE REVIEW

**Comment by darko-poljak** [comment]
@4nd3r  Can you do the removing of legacy code from ssh authorized keys type in separate PR?
Thanks!

**Comment by darko-poljak** [comment]
@4nd3r How about writing stat tool in C, using stat syscall (does Solaris have it?), having C source in __files, uploading it to target, compiling, executing.... :feelsgood:

**Comment by darko-poljak** [comment]
@telmich @asteven Let's give it a try and just merge it?

**Comment by darko-poljak** [comment]
And release 4.11.1.

File: cdist/conf/type/__directory/explorer/stat [hunk 1]
```diff
@@ -39,6 +39,35 @@ group: %g %G
...
       ;;
+    solaris)
+        ls1="$( ls -ld "$destination" )"
+        ls2="$( ls -ldn "$destination" )"
+
+        if [ -f "$__object/parameter/mode" ]
+        then mode_should="$( cat "$__object/parameter/mode" )"
+        fi
+
+        # yes, it is ugly hack, but if you know better way...
```
Comment (telmich) [inline cdist/conf/type/__directory/explorer/stat]: rofl, this is ingenious!

**Comment by telmich** [comment]
good to go

**Pull request merged**